src/com/owncloud/android/services/observer/InstantUploadsObserverBasedOnINotify.java

```diff
@@ -0,0 +1,194 @@
+     * start to record, and saves it empty if the user leaves recording nothing. True store. Life is magic.
+     *
+     * @param event     Kind of event occurred.
+     * @param path      Relative path of the file referred by the event.
+     */
+    @Override
+    public void onEvent(int event, String path) {
+        Log_OC.d(TAG, "Got event " + event + " about "  + ((path != null) ? path : "") +
+            " in current camera folder");
+
+        if (path != null && path.length() > 0) {
+            synchronized (mLock) {
+                Log_OC.d(TAG, "Observed camera folder is " + mConfiguration.getSourcePath());
+
+                if ((event & FileObserver.CREATE) != 0) {
+                    // new file created, let's watch it; false -> not modified yet
+                    mObservedChildren.put(path, false);
+                }
+                if ((   (event & FileObserver.MODIFY) != 0) &&
+                        mObservedChildren.containsKey(path) &&
+                        !mObservedChildren.get(path)
+                    ) {
+                    // watched file was written for the first time after creation
+                    mObservedChildren.put(path, true);
+                }
+                if (   (event & FileObserver.CLOSE_WRITE) != 0 &&
+                        mObservedChildren.containsKey(path)    &&
+                        mObservedChildren.get(path)
+                    ) {
+                    // a file that was previously created and written has been closed;
+                    // testing for FileObserver.MODIFY is needed because some apps
+                    // close the video file right after creating it when the recording
+                    // is started, and reopen it to write with the first chunk of video
+                    // to save; for instance, Camera MX does so.
+                    mObservedChildren.remove(path);
+                    handleNewFile(path);
+                }
+                if ((event & FileObserver.MOVED_TO) != 0) {
+                    // a file has been moved or renamed into the folder;
+                    // for instance, Google Camera does so right after
+                    // saving a video recording
+                    handleNewFile(path);
+                }
+            }
+        }
+
+        if ((event & IN_IGNORE) != 0 &&
+            (path == null || path.length() == 0)) {
+            Log_OC.d(TAG, "Stopping the observance on " + mConfiguration.getSourcePath());
+        }
+    }
+
+
+    /**
+     * Request the upload of a file just created if matches the criteria of the current
+     * configuration for instant uploads.
+     *
+     * @param fileName      Name of the file just created
+     */
+    private void handleNewFile(final String fileName) {
+
+        /// delay a bit the execution to deal with possible renames of files (for instance: Google Camera)
+        mHandler.postDelayed(
```

Comment:
If the delay is removed, the file will be uploaded twice: the first one, when the file is created; the second one, when the file is renamed; both actions would be understood as a different picture, though it's the same. Two requests to ```FileUploader``` would be done. The first one would probably fail because the file wouldn't have the original name when the service really tarts the transfer. The second on would succeed. Both would appear in the uploads view.

The trick here is completed in ```InstantUploadsHandler``` . Before requesting the upload, the handler checks that the file really exists in the path. Without the delay, the check will succeed for the file before the rename. With the delay, when the handler checks the existence the rename already happened, and the first detection will be silently discarded.